src/core/textures/BaseTexture.js

```diff
@@ -405,15 +499,24 @@ BaseTexture.fromImage = function (imageUrl, crossorigin, scaleMode)
             image.crossOrigin = determineCrossOrigin(imageUrl);
         }
 
+        var imageType = getImageTypeOfUrl(imageUrl);
+
+        if (imageType) {
+            image.type = imageType;
```

Comment:
Are the expando properties necessary here? Looks like this is only used in one place, can't the getImageTypeOfUrl call be there instead?